src/smartactions/SchedulingBase.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {IERC7579Execution} from "modulekit/Accounts.sol";
+import {ERC7579ExecutorBase} from "modulekit/Modules.sol";
+import "modulekit/core/sessionKey/ISessionValidationModule.sol";
+
+abstract contract SchedulingBase is
+    ERC7579ExecutorBase,
+    ISessionValidationModule
+{
+    error InvalidExecution();
+    error InvalidMethod(bytes4);
+    error InvalidValue();
+    error InvalidAmount();
+    error InvalidTarget();
+    error InvalidRecipient();
+
+    error InvalidJob();
+
+    event ExecutionAdded(address indexed smartAccount, uint128 indexed jobId);
+
+    event ExecutionTriggered(
+        address indexed smartAccount,
+        uint128 indexed jobId
+    );
+
+    event ExecutionCancelled(
+        address indexed smartAccount,
+        uint128 indexed jobId
+    );
+
+    mapping(address smartAccount => mapping(uint128 jobId => ExecutionConfig))
+        internal _executionLog;
+
+    mapping(address smartAccount => uint128 jobCount) internal _accountJobCount;
+
+    struct ExecutionConfig {
+        uint48 executeInterval;
+        uint16 numberOfExecutions;
+        uint16 numberOfExecutionsCompleted;
+        uint48 startDate;
+        uint160 sqrtPriceLimitX96;
```

Comment:
you could move up the remaining uint48 (and the bool although I'm not 100% sure on this) above the uint160 and then they will all fit into a single storage slot

Reply:
the sqrt is only necessary in case of swaps. so saving it generally doesnt make sense.

why not encoded it into the executionData

Reply:
@zeroknots yeah make sense